packages/react-ui-core/src/DropDownMenu/ItemWrapper.js

```diff
@@ -0,0 +1,52 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import cn from 'classnames'
+import themed from 'react-themed'
+
+@themed(/^DrownDownMenu/, { pure: true })
+
+export default class ItemWrapper extends Component {
+  static propTypes = {
+    children: PropTypes.node,
+    index: PropTypes.number,
+    selectedIndex: PropTypes.number,
+    theme: PropTypes.object,
+    onSelectionHover: PropTypes.func,
+    onSelectionClick: PropTypes.func,
+  }
+
+  constructor(props) {
+    super(props)
+    this.handleHover = this.handleHover.bind(this)
+  }
+
+  handleHover() {
+    this.props.onSelectionHover(this.props.index)
+  }
+
+  render() {
+    const {
+      selectedIndex,
+      index,
+      theme,
+      children,
+      onSelectionClick,
+      ...props
+    } = this.props
+
+    return (
+      <div
+        data-tid="dropdown-menu-list-item"
+        className={cn(
+          theme.DropDownMenu_ListItem,
+          selectedIndex === index && theme.DropDownMenu_Selected
+        )}
+        onClick={onSelectionClick}
+        onMouseEnter={this.handleHover}
```

Comment:
I wonder if we should call `onSelectionHover` `onMouseEnter` so it explains what it's actually doing.  That would also mean you'd have to destructure it out above though.

Reply:
Same as above, we needed it do not have a reserved event function name (`onMouseEnter`) when it is passed to `List` and then passed on to `ItemWrapper` because we didn't want list to inadvertently have event handlers applied to the whole `List` rather than the individual items.

Reply:
Perfectly fine 👍